src/ke.cpp

```diff
@@ -321,6 +324,48 @@ _IRQL_requires_min_(PASSIVE_LEVEL) _When_((timeout == NULL || timeout->QuadPart
             return STATUS_UNSUCCESSFUL;
         }
     }
+    case USERSIM_OBJECT_TYPE_EVENT: {
+        PRKEVENT event = (PRKEVENT)object;
```

Comment:
This block is long enough it would be better to put it in a helper function instead of inline inside the case.